src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -194,168 +185,25 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
-                    Value value = jcrProperty.value
-                    propertyBuilder.setValue(NodeProtos.Value.newBuilder().setStringValue(value.string))
-                }
-                else {
-                    Value[] values = jcrProperty.values
-                    Collection<NodeProtos.Value> protoValues = values.collect { Value value ->
-                        NodeProtos.Value.newBuilder().setStringValue(value.string).build()
-                    }
-                    propertyBuilder.setValues(
-                        NodeProtos.Values.newBuilder().addAllValue(protoValues).build()
-                    )
-                }
-                break
-            case PATH:
-                if (!jcrProperty.multiple) {
-                    Value value = jcrProperty.value
-                    propertyBuilder.setValue(NodeProtos.Value.newBuilder().setStringValue(value.string))
-                }
-                else {
-                    Value[] values = jcrProperty.values
-                    Collection<NodeProtos.Value> protoValues = values.collect { Value value ->
-                        NodeProtos.Value.newBuilder().setStringValue(value.string).build()
-                    }
-                    propertyBuilder.setValues(
-                        NodeProtos.Values.newBuilder().addAllValue(protoValues).build()
-                    )
-                }
-                break
-            case PropertyType.URI:
-                if (!jcrProperty.multiple) {
-                    Value value = jcrProperty.value
-                    propertyBuilder.setValue(NodeProtos.Value.newBuilder().setStringValue(value.string))
-                }
-                else {
-                    Value[] values = jcrProperty.values
-                    Collection<NodeProtos.Value> protoValues = values.collect { Value value ->
-                        NodeProtos.Value.newBuilder().setStringValue(value.string).build()
-                    }
-                    propertyBuilder.setValues(
-                        NodeProtos.Values.newBuilder().addAllValue(protoValues).build()
-                    )
-                }
-                break
-        //TODO: Is it correct to ignore this? (as Reference value would mean different for Server and for Client
-            case REFERENCE:
-                break
-        //TODO: Is it correct to ignore this? (seems similar to REFERENCE to me)
-            case WEAKREFERENCE:
-                break
+        if(jcrProperty.type == BINARY) {
+            propertyBuilder.addValues(valueBuilder.setBytesValue(ByteString.copyFrom(jcrProperty.value.binary.stream.bytes)))
+        }
+        else {
+            //Other property types can potentially have multiple values
+            final Value[] values = jcrProperty.multiple ? jcrProperty.values : [jcrProperty.value] as Value[]
+            values.each { Value value ->
+                propertyBuilder.addValues(valueBuilder.setStringValue(value.string))
```

Comment:
Does that mean it automatically does toString part for all the property types?...
Moreover, Date type property was handled differently in switch case above.. Should we be handling that differently here to?